src/Metadata/PropertyTypeIterable.php

```diff
@@ -0,0 +1,171 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Liip\MetadataParser\Metadata;
+
+use Doctrine\Common\Collections\Collection;
+
+/**
+ * This property type can be merged with PropertyTypeClass<T>, provided that T is, inherits from, or is a parent class of $this->collectionClass
+ * This property type can be merged with PropertyTypeIterable, if :
+ *  - we're not merging a plain array PropertyTypeIterable into a hashmap one,
+ *  - and the collection classes of each are either not present on both sides, or are the same, or parent-child of one another
+ */
+class PropertyTypeIterable extends AbstractPropertyType
```

Comment:
we should make this class final. if inverse what extends what we can do that, and have the deprecated class no longer be technically final.